rust_src/src/buffers.rs

```diff
@@ -129,3 +131,26 @@ pub fn buffer_file_name(buffer: LispObject) -> LispObject {
 
     LispObject::from_raw(buf.filename)
 }
+
+
+#[no_mangle]
+pub extern "C" fn validate_region(b: *mut Lisp_Object, e: *mut Lisp_Object) {
+    let start = LispObject::from_raw(unsafe { *b });
+    let stop = LispObject::from_raw(unsafe { *e });
+
+    let mut beg = start.as_fixnum_coerce_marker_or_error();
```

Comment:
You're still not coercing the original pointers, or swapping the original pointers.